### PR TITLE
[Merged by Bors] - Slight perf improvements and tidy for contributors example

### PR DESCRIPTION
**Pull request by SUPERCILEX**

### CODE REVIEW

File: examples/2d/contributors.rs
Comment (alice-i-cecile) [inline examples/2d/contributors.rs:262]: Because this isn't a proper benchmark, I would probably avoid making this change to prioritize readability.

Comment (alice-i-cecile) [reply examples/2d/contributors.rs:262]: Ditto below.

**Comment by alice-i-cecile** [comment]
I generally really like these changes! They clean things up a bit, and are well-motivated. In real code, I would probably prefer the `for_each` pattern for perf, but the for loop is easier to read for beginners (and Cart has asked us to use that style in examples).

File: examples/2d/contributors.rs
Comment (SUPERCILEX) [reply examples/2d/contributors.rs]: Sounds good, done!

**Review by alice-i-cecile — APPROVED** [review]

**Comment by SUPERCILEX** [comment]
@alice-i-cecile Is there anything I need to do after the approval?

**Comment by alice-i-cecile** [comment]
> @alice-i-cecile Is there anything I need to do after the approval?
Nope, you're all good. If you'd like, you can ask around and wrangle up another approval from the community. Once that's in place, I can merge this in.

(I have merge rights for docs and examples).

**Comment by SUPERCILEX** [comment]
Makes sense, thanks!

File: examples/2d/contributors.rs
Comment (rparrett) [inline examples/2d/contributors.rs:172]: Was your intention here to only update `text.sections[0]` once, rather than every time a new contributor is selected?

If so, `timer_finished()` resets after `.tick()` so I don't think this is doing what you want.

File: examples/2d/contributors.rs
Comment (rparrett) [inline examples/2d/contributors.rs:206]: Following the same logic as the `for_each` discussion above, I'm not sure if we should do this in example code.

File: examples/2d/contributors.rs
Comment (SUPERCILEX) [reply examples/2d/contributors.rs]: Dang, you're right. I used a local to fix this, please let me know if there's a more idiomatic way of doing this.

**Comment by rparrett** [comment]
While we're at it, it would be a bit more idiomatic for that timer to be a resource rather than a component.

File: examples/2d/contributors.rs
Comment (SUPERCILEX) [reply examples/2d/contributors.rs:209]: This is really just a clippy bug, so I don't think the code is any more difficult to understand. On that note, I filed https://github.com/rust-lang/rust-clippy/issues/8361

**Comment by SUPERCILEX** [comment]
@rparrett Like this? 52d57fa0804bde772a9e83d83357f4f2160de4fd

**Comment by rparrett** [comment]
> @rparrett Like this? [52d57fa](https://github.com/bevyengine/bevy/commit/52d57fa0804bde772a9e83d83357f4f2160de4fd)

That looks a lot nicer!

**Review by alice-i-cecile — APPROVED** [review]
I like the further changes! A bit iffy on the clippy ignore, but I agree that it's a clippy bug.

**Comment by SUPERCILEX** [comment]
@alice-i-cecile @rparrett Looks like this should finally be ready for a merge. Any outstanding concerns?

**Comment by rparrett** [comment]
Just a nit, but a few other examples ([plugin](https://github.com/bevyengine/bevy/blob/main/examples/app/plugin.rs#L35), [font_atlas_debug](https://github.com/bevyengine/bevy/blob/main/examples/ui/font_atlas_debug.rs#L17)) use a naming scheme that would look like

```rust
struct SelectionState {
    timer: Timer,
    has_triggered: bool,
}
```

When combining a timer with some other state in a resource.

**Comment by SUPERCILEX** [comment]
Sounds good, done!

**Review by rparrett — APPROVED** [review]
This changeset seems like an improvement to me. I tested with an artificially shortened contributor list to ensure that the wrapping still works.

**Review by alice-i-cecile — APPROVED** [review]

**Comment by SUPERCILEX** [comment]
Woohoo, thanks!

**Pull request closed without merging**
